### PR TITLE
Login, register, authentication

### PR DESCRIPTION
**Pull request by Patrykaca**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,9 @@
     "socket.io-client": "2.4.0",
     "underscore": "^1.13.0",
     "uuid": "^8.3.2",
+    "vee-validate": "^2.2.15",
```
Comment (bartosz-kepka) [inline package.json]: I don't see any usages of this library. Remove it. Anyway, for validation use **vuelidate**. Don't forget to run **npm install** later.

File: package.json [hunk 2]
```diff
@@ -15,7 +15,9 @@
...
     "vue": "^2.6.11",
+    "vue-cookies": "^1.7.4",
```
Comment (bartosz-kepka) [inline package.json]: I don't see any usages of this library. If it's not being used, remove it. Don't forget to run **npm install** later.

File: src/components/global/Navigation.vue [hunk 3]
```diff
@@ -12,6 +12,7 @@
       />
       <v-toolbar-title>{{ $t('navigation.app') }}</v-toolbar-title>
       <v-spacer />
+      <v-spacer />
```
Comment (bartosz-kepka) [inline src/components/global/Navigation.vue]: Unnecessary spacer

File: src/components/global/Navigation.vue [hunk 4]
```diff
@@ -137,10 +138,20 @@ export default {
...
   methods: {
+
+    move_to_register() {
+      console.log('11')
+      this.drawer = false;
+      this.$store.dispatch('user/clearUser');
+      this.$router.push({ name: 'Register' });
+      console.log('22')
+    },
+
```
Comment (bartosz-kepka) [inline src/components/global/Navigation.vue]: Unused method

File: src/components/global/Navigation.vue [hunk 5]
```diff
@@ -137,10 +138,20 @@ export default {
...
     logOut() {
       this.drawer = false;
       this.$store.dispatch('user/clearUser');
-      this.$router.push({ name: 'Login' });
+      //this.$router.push({ name: 'Login' });
+      window.location.reload(true);
```
Comment (bartosz-kepka) [inline src/components/global/Navigation.vue]: Remove comment
If you need to use **window.location.reload(true);** then in most cases something is bugged/broken. Remove it

File: src/locales/en.json [hunk 6]
```diff
@@ -20,10 +20,19 @@
...
     "log-in": "Log in",
+    "register": "Register",
     "error-name-min-length": "Name must be at least 4 characters long",
-    "error-name-required": "Name is required"
+    "error-email": "Wrong email",
+    "error-password-min-length": "Password must be at least 6 characters long",
+    "error-name-required": "Name is required",
+    "error-password-required": "Password is required",
+    "error-login": "Invalid username or password"
+
```
Comment (bartosz-kepka) [inline src/locales/en.json]: Remove blank line

File: src/locales/en.json [hunk 7]
```diff
@@ -20,10 +20,19 @@
   },
   "login": {
     "title": "Please log in",
+    "registerTitle": "Please sign in",
```
Comment (bartosz-kepka) [inline src/locales/en.json]: use kebab-case

File: src/locales/pl.json [hunk 8]
```diff
@@ -20,10 +20,18 @@
   },
   "login": {
     "title": "Zaloguj się",
+    "registerTitle": "Zarejestruj się",
```
Comment (bartosz-kepka) [inline src/locales/pl.json]: as above

File: src/main.js [hunk 9]
```diff
@@ -4,6 +4,9 @@ import VueVirtualScroller from 'vue-virtual-scroller';
 import 'vue-virtual-scroller/dist/vue-virtual-scroller.css';
 import DisableAutocomplete from 'vue-disable-autocomplete';
 
+import VeeValidate from 'vee-validate';
+import VueCookies  from 'vue-cookies';
+
```
Comment (bartosz-kepka) [inline src/main.js]: Remove if unsued

File: src/main.js [hunk 10]
```diff
@@ -20,6 +23,8 @@ const socket = io('http://localhost:8180/', {
 
 Vue.config.productionTip = false;
 Vue.use(Vuelidate);
+Vue.use(VeeValidate);
+Vue.use(VueCookies);
```
Comment (bartosz-kepka) [inline src/main.js]: Remove if unsued

File: src/services/auth-header.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+export default function authHeader() {
+    let user = JSON.parse(localStorage.getItem('user'));
+
+    if (user && user.accessToken) {
+        return { Authorization: 'Bearer ' + user.accessToken }; // for Spring Boot back-end
+        // return { 'x-access-token': user.accessToken };       // for Node.js Express back-end
+    } else {
+        return {};
+    }
+}
```
Comment (bartosz-kepka) [inline src/services/auth-header.js]: Get token from vuex store using getter.
Use localstorage only for storing it between sessions, page refreshes etc. 
Save user info in localstorage when loggin in, clear it when logggin out, read it when app is launching (see main.js  line 33)

File: src/router/index.js [hunk 12]
```diff
@@ -50,15 +56,22 @@ const router = new VueRouter({
...
+   // return;
+  }
+  if (to.name === 'Register' && isAuthenticated) {
+   // next({ name: 'Home' });
+   // return;
   }
   next();
-})
+}
+)
```
Comment (bartosz-kepka) [inline src/router/index.js]: Fix it pls... It cannot be considered done.
```
const publicRoutes = ['Login', 'Register'];

router.beforeEach(({ name }, from, next) => {
    const isAuthenticated = store.getters['user/isAuthenticated'];
    const toPrivateRoute = !publicRoutes.includes(name);

    if (toPrivateRoute && !isAuthenticated) {
      next({ name: 'Login' });
      return;
    }

    if (!toPrivateRoute && isAuthenticated) {
      next({ name: 'Home' });
      return;
    }

    next();
  },
);
```

File: src/services/auth.service.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import axios from 'axios';
+
+const API_URL = 'http://localhost:8080/api/api/auth/';
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]:
```suggestion
const baseURL = '/auth';
```

File: src/services/auth.service.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
+import axios from 'axios';
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]: **take a look at chat.service.js**

```suggestion
import $axios from '@/axios';
```

File: src/services/auth.service.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+const API_URL = 'http://localhost:8080/api/api/auth/';
+
+class AuthService {
+    login(user) {
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]:
```suggestion
    login(credentials) {
```

File: src/services/auth.service.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+        return axios
+            .post(API_URL + 'signin', {
+                username: user.username,
+                password: user.password
+            })
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]:
```suggestion
        const url = `${baseURL}/signin`;
        return axios.post(url, credentials)
```

File: src/services/auth.service.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+            .then(response => {
+                if (response.data.accessToken) {
+                    localStorage.setItem('user', JSON.stringify(response.data));
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]: do it in vuex action

File: src/services/auth.service.js [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+                }
+
+                return response.data;
+            });
+    }
+
+    logout() {
+        localStorage.removeItem('user');
+    }
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]: clear user info from localstorage in vuex action together with vuex state

File: src/services/auth.service.js [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+
+    register(user) {
+        return axios.post(API_URL + 'signup', {
+            username: user.username,
+            email: user.email,
+            password: user.password
+        });
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]:
```suggestion
        const url = `${baseURL}/signup`;
        return axios.post(url, registrationData);
```

File: src/services/auth.service.js [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+    }
+}
+
+export default new AuthService();
```
Comment (bartosz-kepka) [inline src/services/auth.service.js]: make this service an object and move to http subdirectory
```suggestion
export default AuthService;
```

File: src/services/user.service.js [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+    getModeratorBoard() {
+        return axios.get(API_URL + 'mod', { headers: authHeader() });
+    }
+
+    getAdminBoard() {
+        return axios.get(API_URL + 'admin', { headers: authHeader() });
+    }
+}
+
+export default new UserService();
```
Comment (bartosz-kepka) [inline src/services/user.service.js]: remove

File: src/store/auth.module.js [hunk 22]
```diff
@@ -0,0 +1,61 @@
+import AuthService from '../services/auth.service';
+
+const user = JSON.parse(localStorage.getItem('user'));
+const initialState = user
+    ? { status: { loggedIn: true }, user }
+    : { status: { loggedIn: false }, user: null };
```
Comment (bartosz-kepka) [inline src/store/auth.module.js]: load user info from localstorage in load-data.js

File: src/store/auth.module.js [hunk 23]
```diff
@@ -0,0 +1,61 @@
+import AuthService from '../services/auth.service';
```
Comment (bartosz-kepka) [inline src/store/auth.module.js]: dont use relative paths for imports
```suggestion
import AuthService from '@/services/http/auth.service';
```

File: src/store/auth.module.js [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+            state.status.loggedIn = false;
+            state.user = null;
+        },
+        registerSuccess(state) {
+            state.status.loggedIn = false;
+        },
+        registerFailure(state) {
+            state.status.loggedIn = false;
+        }
+    }
```
Comment (bartosz-kepka) [inline src/store/auth.module.js]: Move this logic to src/store/modules/user.js. Analyze it and adjust the code above. Delete this file

File: src/store/index.js [hunk 25]
```diff
@@ -2,6 +2,7 @@ import Vue from 'vue';
 import Vuex from 'vuex';
 
 import user from '@/store/modules/user';
+import { auth } from './auth.module';
```
Comment (bartosz-kepka) [inline src/store/index.js]: this module will be removed

File: src/store/index.js [hunk 26]
```diff
@@ -12,6 +13,7 @@ export default new Vuex.Store({
   mutations: {},
   actions: {},
   modules: {
+    auth,
```
Comment (bartosz-kepka) [inline src/store/index.js]: this module will be removed

File: src/store/modules/user.js [hunk 27]
```diff
@@ -1,6 +1,8 @@
 import i18n from '@/locales/i18n';
 import vuetify from '@/plugins/vuetify';
 
+
+
```
Comment (bartosz-kepka) [inline src/store/modules/user.js]: .

File: src/store/modules/user.js [hunk 28]
```diff
@@ -54,6 +56,9 @@ export default {
     clearUser({ commit }) {
       commit('clearUser');
       localStorage.removeItem('user');
+      //this.$cookies.remove('io', '/socket.io', 'localhost');
+     // document.cookie.sub();
+      window.location.reload(true);
```
Comment (bartosz-kepka) [inline src/store/modules/user.js]: remove unused code, dont reaload the app like that

File: src/store/user.js [hunk 29]
```diff
@@ -0,0 +1,7 @@
+export default class User {
+    constructor(username, email, password) {
+        this.username = username;
+        this.email = email;
+        this.password = password;
+    }
+}
```
Comment (bartosz-kepka) [inline src/store/user.js]: no need to define a class

File: src/views/Login.vue [hunk 30]
```diff
@@ -1,56 +1,93 @@
...
+        {{ $t('login.register') }}
+      </v-btn>
+    </v-form>
+  </div>
 </template>
 
 <script>
-import { minLength, required } from 'vuelidate/lib/validators';
+import {minLength, required} from 'vuelidate/lib/validators';
+import User from '../store/user';
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: don't use relative imports

File: src/views/Login.vue [hunk 31]
```diff
@@ -1,56 +1,93 @@
...
       name: '',
+      user: new User('', ''),
+      loading: false,
+      message: '',
     };
   },
   computed: {
+
+    loggedIn() {
+      return this.$store.state.auth.status.loggedIn;
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: use getter and map it, see ChatConversation.vue line 17

File: src/views/Login.vue [hunk 32]
```diff
@@ -1,56 +1,93 @@
...
+    },
+
+  //  created() {
+  //    if (this.loggedIn) {
+  //      this.$router.push('/profile');
+  //    }
+  //  },
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: remove commented code

File: src/views/Login.vue [hunk 33]
```diff
@@ -64,8 +101,60 @@ export default {
       }
       return errors;
     },
+
+    passwordErrors() {
+      const errors = [];
+      if (!this.$v.name.$dirty) {
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: not **name** but **password**

File: src/views/Login.vue [hunk 34]
```diff
@@ -64,8 +101,60 @@ export default {
...
+        return errors;
+      }
+      if (!this.$v.name.minLength) {
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: not **name** but **password**

File: src/views/Login.vue [hunk 35]
```diff
@@ -64,8 +101,60 @@ export default {
...
+        errors.push(this.$t('login.error-password-min-length'));
+      }
+      if (!this.$v.name.required) {
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: not **name** but **password**

File: src/views/Login.vue [hunk 36]
```diff
@@ -64,8 +101,60 @@ export default {
...
   methods: {
+
+    handleLogin() {
+
+      this.loading = true;
+      this.$validator.validateAll().then(isValid => {
+            if (!isValid) {
+              this.loading = false;
+              return;
+            }
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: use vuelidate

File: src/views/Login.vue [hunk 37]
```diff
@@ -64,8 +101,60 @@ export default {
...
+        errors.push(this.$t('login.error-password-required'));
+      }
+      return errors;
+    },
   },
   methods: {
+
+    handleLogin() {
+
+      this.loading = true;
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: mesage loading in vuex store, see src/store/modules/chat.js method at line 99

File: src/views/Login.vue [hunk 38]
```diff
@@ -64,8 +101,60 @@ export default {
...
+
+        if (this.user.username && this.user.password) {
+
+          this.$store.dispatch('auth/login', this.user).then(
+              () => {
+
+                const user = {
+                  name: this.user.username,
+                };
+                this.$store.dispatch('user/setUser', user);
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: update setUser action from store and dipatch it from login action when user is successfully loggen in

File: src/views/Login.vue [hunk 39]
```diff
@@ -64,8 +101,60 @@ export default {
...
+                this.$router.push({name: 'Home'});
+              },
+              error => {
+                this.loading = false;
+                this.message =
+                    (error.response && error.response.data) ||
+                    error.message ||
+                    error.toString();
+                alert(error.response.data.message);
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: use https://vuetifyjs.com/en/components/snackbars/

File: src/views/Login.vue [hunk 40]
```diff
@@ -64,8 +101,60 @@ export default {
...
+                };
+                this.$store.dispatch('user/setUser', user);
+                this.$router.push({name: 'Home'});
+              },
+              error => {
+                this.loading = false;
+                this.message =
+                    (error.response && error.response.data) ||
+                    error.message ||
+                    error.toString();
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: display custom message from BE

File: src/views/Login.vue [hunk 41]
```diff
@@ -64,8 +101,60 @@ export default {
...
+    handleLogin() {
+
+      this.loading = true;
+      this.$validator.validateAll().then(isValid => {
+            if (!isValid) {
+              this.loading = false;
+              return;
+            }
+
+        if (this.user.username && this.user.password) {
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: validate with vuelidate

File: src/views/Login.vue [hunk 42]
```diff
@@ -64,8 +101,60 @@ export default {
...
+                    error.toString();
+                alert(error.response.data.message);
+              }
+          );
+        }
+      });
+    },
+
+    register() {
+      this.$router.push('/register');
```
Comment (bartosz-kepka) [inline src/views/Login.vue]: use name
```suggestion
            this.$router.push({name: 'Register'});
```

File: src/views/Profile.vue [hunk 43]
```diff
@@ -1,10 +1,38 @@
...
   name: 'Profile',
+  computed: {
+    currentUser() {
+      return this.$store.state.auth.user;
+    }
+  },
+  mounted() {
+    if (!this.currentUser) {
+      this.$router.push('/login');
+    }
```
Comment (bartosz-kepka) [inline src/views/Profile.vue]: route guard should be defined in  src/router/index.js

File: src/views/Profile.vue [hunk 44]
```diff
@@ -1,10 +1,38 @@
...
+  </div>
 </template>
 
 <script>
 export default {
   name: 'Profile',
+  computed: {
+    currentUser() {
+      return this.$store.state.auth.user;
+    }
```
Comment (bartosz-kepka) [inline src/views/Profile.vue]: use getter

File: src/views/Profile.vue [hunk 45]
```diff
@@ -1,10 +1,38 @@
...
+    </p>
+    <p>
+      <strong>Id:</strong>
+      {{ currentUser.id }}
+    </p>
+    <p>
+      <strong>Email:</strong>
+      {{ currentUser.email }}
+    </p>
+  </div>
```
Comment (bartosz-kepka) [inline src/views/Profile.vue]: Expand it, give user a possibility to edit his information

File: src/views/Register.vue [hunk 46]
```diff
@@ -0,0 +1,214 @@
...
+        return errors;
+      }
+      if (!this.$v.name.minLength) {
+        errors.push(this.$t('login.error-password-min-length'));
+      }
+      if (!this.$v.name.required) {
+        errors.push(this.$t('login.error-password-required'));
+      }
+      return errors;
+    },
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: all of them validate only name

File: src/views/Register.vue [hunk 47]
```diff
@@ -0,0 +1,214 @@
...
+
+    loggedIn() {
+      return this.$store.state.auth.status.loggedIn;
+    },
+
+    //  created() {
+    //   if (this.loggedIn) {
+    //     this.$router.push('/profile');
+    //   }
+    // },
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: remove

File: src/views/Register.vue [hunk 48]
```diff
@@ -0,0 +1,214 @@
...
+      user: new User('', ''),
+      submitted: false,
+      successful: false,
+      message: '',
+    };
+  },
+  computed: {
+
+    loggedIn() {
+      return this.$store.state.auth.status.loggedIn;
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: use getter and map it

File: src/views/Register.vue [hunk 49]
```diff
@@ -0,0 +1,214 @@
...
+                alert(this.$t('login.error-login'));
+              }
+          );
+        }
+      });
+    },
+
+    handleRegister() {
+      this.message = '';
+      this.submitted = true;
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: what is that for?

File: src/views/Register.vue [hunk 50]
```diff
@@ -0,0 +1,214 @@
...
+      this.$validator.validate().then(isValid => {
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: use vuelidate

File: src/views/Register.vue [hunk 51]
```diff
@@ -0,0 +1,214 @@
...
+        console.log('chuj');
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: xd

File: src/views/Register.vue [hunk 52]
```diff
@@ -0,0 +1,214 @@
...
+        if (isValid) {
+          this.$store.dispatch('auth/register', this.user).then(
+              data => {
+                this.message = data.message;
+                this.successful = true;
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: what is that for?

File: src/views/Register.vue [hunk 53]
```diff
@@ -0,0 +1,214 @@
...
+                this.handleLogin();
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: redirect to login page

File: src/views/Register.vue [hunk 54]
```diff
@@ -0,0 +1,214 @@
...
+
+    handleRegister() {
+      this.message = '';
+      this.submitted = true;
+      this.$validator.validate().then(isValid => {
+        console.log('chuj');
+        if (isValid) {
+          this.$store.dispatch('auth/register', this.user).then(
+              data => {
+                this.message = data.message;
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: show a notification using notification module of vuex store, see usage at chat module

File: src/views/Register.vue [hunk 55]
```diff
@@ -0,0 +1,214 @@
...
+                this.successful = true;
+                this.handleLogin();
+              },
+              error => {
+                alert(error.response.data.message);
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: use snackbar from vuetify

File: src/views/Register.vue [hunk 56]
```diff
@@ -0,0 +1,214 @@
...
+                this.message =
+                    (error.response && error.response.data) ||
+                    error.message ||
+                    error.toString();
+                alert(this.$t('login.error-login'));
+              }
+          );
+        }
+      });
+    },
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: remove

File: src/views/Register.vue [hunk 57]
```diff
@@ -0,0 +1,214 @@
...
+
+export default {
+  name: 'Register',
+  data() {
+    return {
+      name: '',
+      user: new User('', ''),
+      submitted: false,
+      successful: false,
+      message: '',
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: what is that for? where are you using it?

File: src/views/Register.vue [hunk 58]
```diff
@@ -0,0 +1,214 @@
...
+      this.$store.dispatch('user/setUser', user);
+      this.$router.push({name: 'Home'});
+    },
+  },
+  validations: {
+    name: {
+      required,
+      minLength: minLength(4),
+    },
+  },
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: define validations for input fields here

File: src/views/Register.vue [hunk 59]
```diff
@@ -0,0 +1,214 @@
...
+      if (this.$v.$invalid) {
+        return;
+      }
+      const user = {
+        name: this.name,
+      };
+      this.$store.dispatch('user/setUser', user);
+      this.$router.push({name: 'Home'});
+    },
+  },
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: login?

File: src/views/Register.vue [hunk 60]
```diff
@@ -0,0 +1,214 @@
...
+        lg="1"
+        md="3"
+        sm="3"
+      >
+        <v-btn
+          color="primary"
+          type="submit"
+        >
+          {{ $t('login.register') }}
+        </v-btn>
```
Comment (bartosz-kepka) [inline src/views/Register.vue]: add button for going back bo login page

File: src/views/Login.vue [hunk 61]
```diff
@@ -75,7 +164,7 @@ export default {
         name: this.name,
       };
       this.$store.dispatch('user/setUser', user);
-      this.$router.push({ name: 'Home' });
+      this.$router.push({name: 'Home'});
     },
   },
   validations: {
```
Comment (bartosz-kepka) [inline src/views/Login.vue:105]: define validations for input fields here

**Review by bartosz-kepka — CHANGES_REQUESTED** [review]
fix pls
use http interceptor fot adding authorization header to requests which need it

File: package.json (hunk 1 above)
Comment (Patrykaca) [reply package.json]: after deleting login button stops working

File: package.json (hunk 2 above)
Comment (Patrykaca) [reply package.json]: after deleting login button stops working

File: src/store/auth.module.js (hunk 23 above)
Comment (Patrykaca) [reply src/store/auth.module.js]: Cannot install Node.js module: "C:\Program Files\nodejs\node.exe" "C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js" install @/services --save-dev

File: package.json (hunk 1 above)
Comment (bartosz-kepka) [reply package.json]: My bad comment, this library is used for validation while logging in. Please replace its usage with vuelidate as mentioned here: https://github.com/bartosz-kepka/social-media-app/pull/1/files#r680252017

File: package.json (hunk 2 above)
Comment (bartosz-kepka) [reply package.json]: Same as above

File: src/store/modules/user.js (hunk 28 above)
Comment (Patrykaca) [reply src/store/modules/user.js]: i don't have any idea how to log out a different way properly, without it cookies are not removed (I think)

File: src/views/Register.vue (hunk 60 above)
Comment (Patrykaca) [reply src/views/Register.vue]: done

File: src/store/auth.module.js (hunk 23 above)
Comment (bartosz-kepka) [reply src/store/auth.module.js]: It's not a module to install. It's a path relative to src directory: https://stackoverflow.com/a/42753045/10876483

File: src/store/modules/user.js (hunk 28 above)
Comment (bartosz-kepka) [reply src/store/modules/user.js]: User-specific info like authentication token and roles should be stored in vuex store and local storage. Local storage should be a copy of info stored in vuex. We consider a user to be loggen in when there are some information saved in vuex store so "to log out" means to clear all user info from both vuex store and local storage. A copy of user info should be stored in local storage so that we can load it after page refresh or when opening a website later (local storage is persistent while every time we open an app we create a new instance of vuex store this is empty). Create a getter which  would indicate whether some user is logged in (token, login, roles etc. are saved in vuex store).
Take a look here: https://github.com/bartosz-kepka/social-media-app/pull/1#discussion_r680227894 -**user/isAuthenticated** getter is used to specify if user some user is logged in. 

To sum up:
1. Store currently logged in user info in vuex
2. Create a getter which will indicate whether some user is logged in
3. When "Log out" button (or similar) is clicked then dispatch an action that will remove user info from vuex store (from local storage too as it should be a copy)
4. Because we cleared user info the getter should indicate that there is no logged in user - e.g. getter should return false now (instead of true)
5. Redirect to login page

**Review by bartosz-kepka — APPROVED** [review]
